jormungandr/src/settings/start/mod.rs

```diff
@@ -124,28 +125,54 @@ impl RawSettings {
         };
 
         let explorer = command_arguments.explorer_enabled
-            || config.explorer.map_or(false, |settings| settings.enabled);
+            || config.as_ref().map_or(false, |cfg| {
+                cfg.explorer
+                    .as_ref()
+                    .map_or(false, |settings| settings.enabled)
+            });
 
         Ok(Settings {
             storage: storage,
             block_0: block0_info,
             network: network,
             secrets,
-            rest: config.rest,
-            mempool: config.mempool,
-            leadership: config.leadership,
+            rest: config.as_ref().map_or(None, |cfg| cfg.rest.clone()),
+            mempool: config
+                .as_ref()
+                .map_or(Mempool::default(), |cfg| cfg.mempool.clone()),
+            leadership: config
+                .as_ref()
+                .map_or(Leadership::default(), |cfg| cfg.leadership.clone()),
             explorer,
         })
     }
 }
 
 fn generate_network(
-    _command_arguments: &StartArguments,
-    config: &Config,
+    command_arguments: &StartArguments,
+    config: &Option<Config>,
```

Comment:
Consider initializing `Config` with default values in case the file is missing, rather than dragging the `Option` through the APIs.